rsync_tmbackup.sh

```diff
@@ -546,9 +546,9 @@ while : ; do
 	if [ -n "$SSH_CMD" ]; then
 		RSYNC_FLAGS="$RSYNC_FLAGS --compress"
 		if [ -n "$ID_RSA" ] ; then
-			CMD="$CMD  -e 'ssh -p $SSH_PORT -i $ID_RSA -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null'"
+			CMD="$CMD  -e 'ssh -p $SSH_PORT -i $ID_RSA -o StrictHostKeyChecking=no'"
```

Comment:
This is line is already multiple security flaws in one, namely: CWE-547, CWE-347 and CWE-1188.
if the environment is unable to use ssh key checking properly, WHY are you even using ssh? 
TL;DR; all you gain is a slower backup that is also weakening your encryption (see line 547 ... encryption is the opposite of compression ... when security matters avoid using both at the same time, it's speed vs security, not both)
at the very least these exceptions belong in the `/etc/ssh/ssh_config` file when appropriate. Better to add these to a variable for the time being:
```bash
# insert at 548
# TODO: add a --no-key-checks (or --cron) option to this tool
SSH_ARGS="-o StrictHostKeyChecking=no"
if [ -n "$ID_RSA" ] ; then
SSH_ARGS="-i $ID_RSA $SSH_ARGS"
fi
CMD="$CMD  -e 'ssh -p $SSH_PORT $SSH_ARGS"
# continue at line 552
```
this allows for the current code's function and allows other code to cope with security, excusing this code from the issue without taking it on all at once.

Reply:
I agree with you but this PR is just regarding cron friendliness and not about security. Another PR would be more appropriate for such a change.

Reply:
> I agree with you but this PR is just regarding cron friendliness and not about security. Another PR would be more appropriate for such a change.

😕 opps! sorry ✔️ I agree with your second point the ...
> This is line is already multiple security flaws in one, namely: CWE-547, CWE-347 and CWE-1188.
... should be a separate issue or PR ...

✔️ I get this was about #128 (and thus also #104) _only_ as they relate to cron usage ... correct me if I'm wrong.

I agree with 
> this PR is just regarding cron friendliness and not about security
but I still hold what I said
> Better to add these to a variable for the time being:
> 
> ```shell
> # insert at 548
> # TODO: add a --no-key-checks (or --cron) option to this tool
> SSH_ARGS="-o StrictHostKeyChecking=no"
> if [ -n "$ID_RSA" ] ; then
> SSH_ARGS="-i $ID_RSA $SSH_ARGS"
> fi
> CMD="$CMD  -e 'ssh -p $SSH_PORT $SSH_ARGS"
> # continue at line 552
> ```
> 
> this allows for the current code's function and allows other code to cope with security, excusing this code from the issue without taking it on all at once.

And your initial comment:

> The second change was the `-o UserKnownHostsFile=/dev/null` fixed ssh argument. This lead to ssh always writing to STDERR. This is part of #128. I would expect people to first setup the ssh access and then use rsync_tmbackup.sh. I cant' see a reason in bypassing the local configuration by default. There is always the `-e` option of rsync which can be used to set ssh options.

so...

I guess the --cron option would be the easiest to maintain, and keeps this clearly out of the _security holy war_. 🤔 ...Thoughts?

Reply:
I'm not sure I get this. Why add a --cron flag? Why should normal use write to STDERR everytime? By redirecting STDOUT to /dev/null you should be able to expect emails (from cron) only when something unexpected happens.